### ISSUE TITLE
Breaks ComboBox /2

### ISSUE THREAD

**Issue by pinonpierre** [vaadin/framework#11343]
Example to reproduce : 
```
 VerticalLayout mainLayout = new VerticalLayout();

        ListDataProvider<String> dataProvider = new ListDataProvider<>(new ArrayList<>());

        Label label = new Label();
        mainLayout.addComponent(label);

        ComboBox<String> comboBox = new ComboBox<>("ComboBox");
        comboBox.setItems();
        comboBox.setDataProvider(dataProvider);
        comboBox.setNewItemProvider(value -> {

            dataProvider.getItems().add(value);
            dataProvider.refreshAll();

            try {
                Thread.sleep(100);
            } catch (InterruptedException e) {
                e.printStackTrace();
            }

            return Optional.ofNullable(value);
        });
        mainLayout.addComponent(comboBox);

        Binder<AtomicReference<String>> binder = new Binder<>();
        binder.forField(comboBox)
                .asRequired("Required")
                .bind(AtomicReference::get, AtomicReference::set);

        Button button = new Button("Clear");
        mainLayout.addComponent(button);
        button.addClickListener(event1 -> {
            comboBox.setValue(null);
            dataProvider.getItems().clear();
            dataProvider.refreshAll();
        });

        Button button2 = new Button("Validate");
        mainLayout.addComponent(button2);
        button2.addClickListener(event1 -> binder.validate());
        setContent(mainLayout);
```
Steps to reproduce: 
- Enter the value "myValue" in combobox field but don't press any key and don't click anywhere
- Click on the "Clear" Button
- Click on the "Validate" Button
- No way to choose the value "myValue" or to add another value and to validate for now

Reopened issue of : https://github.com/vaadin/framework/issues/11059

Reproducible in 8.6.3

**Comment by TatuLund** [comment]
Is this related to https://github.com/vaadin/framework/issues/11157 ?

**Comment by ZheSun88** [comment]
issue fixed and included in Framework 8.8.0
